File: Estructura_de_datos_JavaScript/Hash_Sort.js
function hashSort(arr, maxVal) {
  const hashTable = new Array(maxVal + 1).fill(false);

  arr.forEach(num => hashTable[num] = true);

  const sorted = [];
  for (let i = 0; i <= maxVal; i++)
    if (hashTable[i]) sorted.push(i);

  return sorted;
}

let arr = [5, 2, 8, 3, 1];
console.log(hashSort(arr, 10));

File: Ordenamiento_Python/Hash_Sort.py
def hash_sort(arr, max_val):
    hash_table = [0] * (max_val + 1)

    for num in arr:
        hash_table[num] = 1

    sorted_arr = [i for i in range(len(hash_table)) if hash_table[i] == 1]
    return sorted_arr

arr = [5, 2, 8, 3, 1]
print(hash_sort(arr, 10))
